roles/openshift_logging_curator/tasks/main.yaml

```diff
@@ -92,36 +104,124 @@
+        state: present
+        name: "logging-curator"
+        namespace: "{{ openshift_logging_namespace }}"
+        from_file:
+          config.yaml: "{{ tempdir }}/curator.yml"
+
+    - name: Set Curator DC
+      oc_obj:
+        state: present
+        name: "{{ curator_name }}"
+        namespace: "{{ openshift_logging_namespace }}"
+        kind: dc
+        files:
+        - "{{ tempdir }}/templates/curator-dc.yaml"
+        delete_after: true
+
+# Cron Job - v5.x
+- when: openshift_logging_es5_techpreview | bool
+  block:
+    - name: Generate Curator cronjob
+      template:
+        src: "{{ __base_file_dir }}/curator-cj.j2"
+        dest: "{{ tempdir }}/templates/curator-cj.yaml"
+      vars:
+        component: "{{ curator_component }}"
+        logging_component: curator
+        deploy_name: "{{ curator_name }}"
+        image: "{{openshift_logging_curator_image_prefix}}logging-curator:{{openshift_logging_curator_image_version}}"
+        es_host: "{{ openshift_logging_curator_es_host }}"
+        es_port: "{{ openshift_logging_curator_es_port }}"
+        curator_cpu_limit: "{{ openshift_logging_curator_cpu_limit }}"
+        curator_cpu_request: "{{ openshift_logging_curator_cpu_request | min_cpu(openshift_logging_curator_cpu_limit | default(none)) }}"
+        curator_memory_limit: "{{ openshift_logging_curator_memory_limit }}"
+        curator_node_selector: "{{openshift_logging_curator_nodeselector | default({})}}"
+        cron_job_schedule: "{{ openshift_logging_curator_run_minute | default(0) }} {{ openshift_logging_curator_run_hour | default(0) }} * * *"
+      check_mode: no
+      changed_when: no
+
+    # Copy config files
+    - copy:
+        src: "{{ __base_file_dir }}/actions.yaml"
+        dest: "{{ tempdir }}/actions.yaml"
+    
+    - copy:
+        src: "{{ __base_file_dir }}/config.yaml"
+        dest: "{{ tempdir }}/curator5.yaml"
+    
+    # Patch existing configuration, if present
+    - import_role:
+        name: openshift_logging
+        tasks_from: patch_configmap_files.yaml
+      vars:
+        configmap_name: "logging-curator"
+        configmap_namespace: "logging"
+        configmap_file_names:
+          - current_file: "actions.yaml"
+            new_file: "{{ tempdir }}/actions.yaml"
+          - current_file: "curator5.yaml"
+            new_file: "{{ tempdir }}/curator5.yaml"
+
+    # TODO: handle upgrade
+    - name: Set Curator configmap
+      oc_configmap:
+        state: present
+        name: "logging-curator"
+        namespace: "{{ openshift_logging_namespace }}"
+        from_file:
+          actions.yaml: "{{ tempdir }}/actions.yaml"
+          curator5.yaml: "{{ tempdir }}/curator5.yaml"
+
+    - name: Set Curator Cronjob
+      oc_obj:
+        state: present
+        name: "{{ curator_name }}"
+        namespace: "{{ openshift_logging_namespace }}"
+        kind: cronjob
+        files:
+          - "{{ tempdir }}/templates/curator-cj.yaml"
+        delete_after: true
```

Comment:
aren't we also going to delete/scale down the old curator replica as part of this path?